src/commands/fetch.js

```diff
@@ -175,6 +175,9 @@ async function fetchPackfile ({
     headers
   })
   auth = remoteHTTP.auth // hack to get new credentials from CredentialManager API
+  if (!remoteHTTP.capabilities.size) {
```

Comment:
It looks too much like `capabilities` is a regular Object and we're checking for a `size` capability on the server. Let's make this explicit:
```
// You cannot clone an empty repository.
if (remoteHTTP.capabilities.size === 0) {
```

hang on. Why are we looking at `remoteHTTP.capabilities`? Shouldn't we be looking at `remoteHTTP.refs`?

Reply:
That was exactly my question...and you've answered it. Both are empty in this condition, so I wasn't sure which one to rely on. But I agree refs would be more clear.